### PR TITLE
`Development`: Add playwright e2e tests for logout

### PR DESCRIPTION
**Pull request by muradium**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
Logout tests are missing in Playwright. As a part of migration to Playwright, they need to be migrated from Cypress.

### Description
Adds logout tests to Playwright and disables them in Cypress.

### Steps for Testing
- **Code Review**: Ensure that tests pass for valid reasons, functionality is adequately tested and check the code quality.
- **Run the tests (optional)**: Tests are checked by automatic run in CI environment. You can optionally run them locally and check if they pass.

**Steps for running the tests:**
1. Navigate to **src/tests/playwright**
2. Configure Playwright using playwright.env file based on your local setup. Current configuration should work for default Artemis setup.
3. Run `npm install && npm run playwright:setup && npx playwright test e2e/Logout.spec.ts`
4. Confirm that **"Logout"** tests pass

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

## Summary by CodeRabbit

- **Tests**
	- Enhanced testing for logout scenarios by adding tests for handling unsaved changes, verifying behavior for different user roles, and testing logout confirmation during exercise mode.
	- Improved testing efficiency by temporarily skipping execution of specific logout tests.

### CODE REVIEW

File: src/test/playwright/e2e/Logout.spec.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    test.beforeEach('Login as admin and create a course with a modeling exercise', async ({ login, courseManagementAPIRequests, exerciseAPIRequests }) => {
+        await login(admin);
+
+        course = await courseManagementAPIRequests.createCourse();
+        await courseManagementAPIRequests.addStudentToCourse(course, studentOne);
+        await courseManagementAPIRequests.addStudentToCourse(course, studentTwo);
+        modelingExercise = await exerciseAPIRequests.createModelingExercise({ course });
+    });
+
+    test('Logs out by pressing OK when unsaved changes on exercise mode', async ({ page, login, courseOverview, modelingExerciseEditor, navigationBar }) => {
```
Comment (julian-christl) [inline src/test/playwright/e2e/Logout.spec.ts:20]:
```suggestion
    test('Logs out after confirmation of dialog for unsaved changes on exercise mode', async ({ page, login, courseOverview, modelingExerciseEditor, navigationBar }) => {
```

File: src/test/playwright/e2e/Logout.spec.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            expect(dialog.message()).toContain('You have unsaved changes');
+            await dialog.accept();
+        });
+
+        await navigationBar.logout();
+
+        await page.waitForURL(process.env.BASE_URL!);
+    });
+
+    test('Stays logged in by pressing cancel when trying to logout during unsaved changes on exercise mode', async ({
```
Comment (julian-christl) [inline src/test/playwright/e2e/Logout.spec.ts:40]:
```suggestion
    test('Stays logged after dismissal of dialog for unsaved changes on exercise mode', async ({
```

**Review by julian-christl — COMMENTED** [review]
Makes it a bit cleare what the tests do.

**Review by pzdr7 — APPROVED** [review]
Code looks good. Tests pass locally 👍

**Comment by julian-christl** [comment]
@muradium did you only apply the one suggestion on purpose?

File: src/test/playwright/e2e/Logout.spec.ts (hunk 2 above)
Comment (julian-christl) [reply src/test/playwright/e2e/Logout.spec.ts]: Yes, of course!

**Review by julian-christl — APPROVED** [review]

**Review by pzdr7 — APPROVED** [review]

**Pull request merged**